src/helpers/auth.ts

```diff
@@ -0,0 +1,93 @@
+/**
+ * @license
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export interface ServiceAccount {
+  type: 'this.serviceAccount';
+  project_id: string;
+  private_key_id: string;
+  private_key: string;
+  client_email: string;
+  client_id: string;
+  auth_uri: 'https://accounts.google.com/o/oauth2/auth';
+  token_uri: 'https://oauth2.googleapis.com/token';
+  auth_provider_x509_cert_url: 'https://www.googleapis.com/oauth2/v1/certs';
+  client_x509_cert_url: string;
+  user_email: string;
+}
+
+/**
+ * This is a wrapper class for handling authentification to DV360 API.
+ * This class can be used to auth also to other Google APIs.
+ */
+export class Auth {
```

Comment:
Would it make sense to introduce an explicit mode "USER" or "SERVICE_ACCOUNT" here for developers to choose? Implicitly choosing a mode depending on the presence of properties of the options object may lead to unexpected behavior.